### PR TITLE
V535, V540

### LINKED ISSUE #170: Validation (2017): V540 (HOEPA Status)
**Edit #:** V540
**Type:** Validity
**Category:** Loan/Application Register (only)
**Column:** HOEPA Status

If action taken type = 2-5, 7, or 8, HOEPA status must = 2.

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### LINKED ISSUE #172: Validation (2017): V535 (HOEPA Status)
**Edit #:** V535
**Type:** Validity
**Category:** Loan/Application Register (only)
**Column:** HOEPA Status

If applicant ethnicity, race and sex = 4, 7, and 4, respectively, and action taken does not = 6 HOEPA status must not = 1.

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### PR DESCRIPTION
**Pull request by kgudel**
Closes #170 
Closes #172

### CODE REVIEW

File: validation/src/main/scala/hmda/validation/rules/lar/validity/V535.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import hmda.validation.dsl.PredicateSyntax._
+
+object V535 extends EditCheck[LoanApplicationRegister] {
+
+  override def apply(lar: LoanApplicationRegister): Result = {
+    when(
+      (lar.applicant.coEthnicity is equalTo(4))
+        and (lar.applicant.coRace1 is equalTo(7))
+        and (lar.applicant.coSex is equalTo(4))
+        and (lar.actionTakenType not equalTo(6))
```
Comment (jmarin) [inline validation/src/main/scala/hmda/validation/rules/lar/validity/V535.scala]: I think this checks are not correct, you are checking for co-applicant (i.e. coEthnicity), while `V535` talks about applicant directly.

File: validation/src/test/scala/hmda/validation/rules/lar/validity/V540Spec.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V540Spec extends LarEditCheckSpec with BadValueUtils {
+
+  val badActionTakenTypes = List(2, 3, 4, 5, 7, 8)
+  val validActionTakenGen: Gen[Int] =
```
Comment (jmarin) [inline validation/src/test/scala/hmda/validation/rules/lar/validity/V540Spec.scala]: I suggest not using "Types" in your variable naming moving forward, unless you are doing a `type` alias in `Scala`. I would use something like `badActionTakenValues`

File: validation/src/main/scala/hmda/validation/rules/lar/validity/V540.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.rules.EditCheck
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+
+object V540 extends EditCheck[LoanApplicationRegister] {
+
+  val ActionTakenList = List(2, 3, 4, 5, 7, 8)
```
Comment (jmarin) [inline validation/src/main/scala/hmda/validation/rules/lar/validity/V540.scala]: Small thing, but usually by convention variables follow camel casing with the first letter being lower case (i.e. actionTakenList)

Comment (kgudel) [inline validation/src/main/scala/hmda/validation/rules/lar/validity/V540.scala]: I appreciate the style tip, thanks Juan.

**Comment by schbetsy** [comment]
LGTM

**Pull request merged**